### PR TITLE
reporting code

### PR DESCRIPTION
**Pull request by shannog2**
# Summary

Code to generate stats and report on false calls in varsim simulations. Annotated augmented_fn.vcf.gz and augmented_fp.vcf.gz files are created containing details in the INFO vcf field:

augmented_fp_annotated.vcf.gz INFO field:
AF; truth variant(if applicable); pure fp/not pure fp

augmented_fn_annotated.vcf.gz INFO field:
AF;

### CODE REVIEW

File: compare_vcf.py [hunk 1]
```diff
@@ -398,6 +493,8 @@ def summarize_results(prefix, tp, fn, fp, t, var_types, sv_length = 100, regions
                              choices = ['SNP', 'Deletion', 'Insertion', 'Inversion', 'TandemDup',
                                        'Complex', 'TransDup', 'TansDel', 'InterDup', 'Translocation'], required = False)
     main_parser.add_argument("--true_vcf", metavar="VCF", help="Input small variant sampling VCF, usually dbSNP", required = True)
+    main_parser.add_argument("--master_vcf", metavar="MASTER_VCF", help="Master whitelist, if applicable", required = False)
+    main_parser.add_argument("--call_vcf", metavar="CALL_VCF", help="Original, VCF output by variant caller, if applicable", required = False)
```
Comment (yunfeiguo) [inline compare_vcf.py]: Hi @shannog2 `--vcfs` and `--call_vcf` should be the same thing, and `--vcfs` only supports one input file for now. see https://github.com/bioinform/varsim/blob/d8d262a455e209ade7d5b401fbb3800028e129cd/compare_vcf.py#L237

File: compare_vcf.py [hunk 2]
```diff
@@ -287,10 +287,105 @@ def process(args):
...
+
+                line_split[6] = info
+                annotated_content.append('\t'.join(line_split))
+
+                #clean up
+                if os.path.isfile(single_var_file):
+                    os.remove(single_var_file)
+                    os.remove(single_var_file+".tbi")
+
+    annotated_file = utils.write_vcf(annotated_content, os.path.join(args.out_dir, "{}_annotated.vcf".format(os.path.splitext(os.path.splitext(os.path.basename(augmented_file))[0])[0])))
```
Comment (yunfeiguo) [inline compare_vcf.py]: Why is `splitext` used twice here? Wonder if direct string removal (say `rstrip`) would be simpler?

File: utils.py [hunk 3]
```diff
@@ -176,52 +175,89 @@ def combine_vcf(combined_vcf, vcfs, duplicate_handling_mode = COMBINE_KEEP_ALL_D
...
+
+
+def make_clean_vcf(vcf, path=None):
+    """Make a clean vcf retaining essential fields"""
+
+    vcf_path = os.path.split(vcf)
+    vcf_base = os.path.splitext(vcf_path[1])
+
+    if not path:
+        path = vcf_path
```
Comment (yunfeiguo) [inline utils.py]: 👍

File: utils.py [hunk 4]
```diff
@@ -176,52 +175,89 @@ def combine_vcf(combined_vcf, vcfs, duplicate_handling_mode = COMBINE_KEEP_ALL_D
...
+    with versatile_open(vcf, "r") as vcf_handle:
+        for line in vcf_handle.readlines():
+            line_strip = line.strip()
+            line_split = line_strip.split()
+
+            if line_strip[0] == "#":
+                clean_vcf_handle.write(line_strip+'\n')
+                continue
+            else:
+                GT = "0/1" if line_split[-1] == "./." else line_split[-1]
```
Comment (yunfeiguo) [inline utils.py]: em, are we sure to convert all `./.` to `0/1`?

File: utils.py [hunk 5]
```diff
@@ -176,52 +175,89 @@ def combine_vcf(combined_vcf, vcfs, duplicate_handling_mode = COMBINE_KEEP_ALL_D
...
+        pysam.tabix_index(combined_vcf, force=True, preset='vcf')
         return gz_vcf
     else:
         return combined_vcf
 
-def index_vcf_gz(vcf_gz):
-    pysam.tabix_index(vcf_gz, force = True, preset = 'vcf')
 
-def sort_and_compress(vcf, mode = 1, overwrite = False):
+def sort_and_compress(vcf):
```
Comment (yunfeiguo) [inline utils.py]: Was the updated `utils.py` used here?

File: compare_vcf.py (hunk 2 above)
Comment (shannog2) [reply compare_vcf.py]: Good spot, I'll change this

File: utils.py (hunk 4 above)
Comment (shannog2) [reply utils.py]: If the user doesn't specify GT, shouldn't we should default to something? Do you think there's a better option?

File: compare_vcf.py (hunk 1 above)
Comment (shannog2) [reply compare_vcf.py]: No, apologies for mixing varsim with the carrier screening pipeline, but in the pipeline when using compare_vcf for the whitelist, we have "--vcfs $wl_intersect_vcf" this does not contain the AO and RO information I need, and so this additional parameter was introduced.

We need to decide if annotating the false call vcfs could be useful for generic varsim sims. If yes, having the flexibility these additional parameters provide could be useful. If not we'll have to cancel this pull request and do something specifically for the CS pipeline.

File: utils.py (hunk 5 above)
Comment (shannog2) [reply utils.py]: I see utils.py was updated recently. I will fix this.

**Pull request merged**